waspc/src/Wasp/Lib.hs

```diff
@@ -83,6 +85,14 @@ analyzeWaspProject waspDir options = do
                   AS.isBuild = CompileOptions.isBuild options
                 }
 
+-- | Checks the wasp directory for potential problems, and issues warnings if any are found.
+analyzeWaspDir :: Path' Abs (Dir WaspProjectDir) -> IO [GeneratorWarning]
```

Comment:
This warning doesn't really have anything to do with Generator.
`compile` returns CompileWarning, which is just String, so we should rather return that.

I would also consider doing this check at the very beginning, since it doesn't really depend on any other checks (analyzer, generator, ...), so we can just add it to any other warnings / errors.

So we could do smth like:
```hs
compile waspDir outDir options = do
  waspDirWarnings <- analyzeWaspDir waspDir
  appSpecOrAnalyzerErrors <- analyzeWaspProject waspDir options
  compilerWarningsAndErrors <- case appSpecOrAnalyzerErrors of
    Left analyzerErrors -> return ([], analyzerErrors)
    Right appSpec ->
      case ASV.validateAppSpec appSpec of
        [] -> do
          (generatorWarnings, generatorErrors) <- Generator.writeWebAppCode appSpec outDir (sendMessage options)
          return (map show $ generatorWarningsFilter options generatorWarnings, map show generatorErrors)
        validationErrors -> do
          return ([], map show validationErrors)
   
   return $ compilerWarningsAndErrors <> waspDirWarnings
```

What remains a bit weird is that we have `analyzeWaspDir` and `analyzeWaspProject` -> those sound almost equal to each other.
Maybe that indicates that this check for dot env file should go into `analyzeWaspProject`? We fetch other env files there anyway. So I would consider doing it there, and extending the signature of `analyzeWaspProject` to also return warnings.

Reply:
Ok, I updated `analyzeWaspProject` to return warnings, and I think you may like the signature :D came from our Generator monad transformer stack days lol